### PR TITLE
travis: separate check patch and age of git base

### PR DESCRIPTION
**Pull request by marc-hb**
As requested in fix for checkpatch issue PR #2543 which will be rebased
on top of this.

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -81,6 +85,17 @@ jobs:
...
+    - name: checkpatch
+      script:
+        # If git misses this merge base then 'git rev-list
+        # ^${TRAVIS_BRANCH} ${TRAVIS_PULL_REQUEST_SHA}' will tell
+        # checkpatch to scan practically every single git commit in the
+        # repo.
+        #
+        # FIXME: this addresses only pull requests without any merge
+        # submitted as part of the pull request itself.
+        - *unshallow_for_merge_base
```
Comment (plbossart) [inline .travis.yml:97]: out of curiosity, what are those '*unshallow_for_merge_base" and "&unshallow_for_merge_base"?

Comment (marc-hb) [reply .travis.yml:97]: They're YAML anchors = how to avoid copy/paste/diverge.

**Comment by kv2019i** [comment]
@marc-hb Shouldn't the checkpatch run https://travis-ci.org/github/thesofproject/linux/builds/740309390 succeed here (i.e. detect git clone is too shallow and run unshallow on the repo before checkpatch)...?

**Comment by marc-hb** [comment]
> @marc-hb Shouldn't the checkpatch run succeed here (i.e. detect git clone is too shallow and run unshallow on the repo before checkpatch)...?

Yes it should. But it doesn't. @ranj063 spotted a similar issue in https://travis-ci.org/github/thesofproject/linux/jobs/739977895.

I reproduced this locally at last and root-caused it to the crazy (buggy?) way `git fetch` works in shallow clones. I'm drafting a bug right now to document this.

**Comment by marc-hb** [comment]
Rebased and added `--all` option to `git merge-base` command to help (a bit) with shallowness issues mentioned above.

https://travis-ci.org/github/thesofproject/linux/builds/741026106 is looking good.

Even though the #2543 "fix" is uncertain and needs to be discussed and revisited I think this PR is good because it has no real functional change, it's just removing a confusing "optimization" and making the Travis output simpler and clearer.

PS: ICL_RVP_HDA's failure to boot in  https://sof-ci.01.org/linuxpr/PR2547/build4824/devicetest can obviously not be related to this.

**Comment by marc-hb** [comment]
> I reproduced this locally at last and root-caused it to the crazy (buggy?) way git fetch works in shallow clones. I'm drafting a bug right now to document this.

Finally done, see graphical explanation of the issue in freshly filed `Shallow cloning in CI can create pull requests with one million commits` #2556

**Comment by plbossart** [comment]
Not sure where this is going and if Travis is still relevant. closing.

**Comment by marc-hb** [comment]
Summary at https://github.com/thesofproject/linux/pull/2543#issuecomment-765029638

**Pull request closed without merging**